src/cow-react/modules/fortune/containers/FortuneWidget/index.tsx

```diff
@@ -1,156 +1,321 @@
+    )
+  }, [lastCheckedFortune])
 
-    if (lastFortuneWasOpenDayAgo) {
-      showFortuneButton()
+  const closeModal = useCallback(() => {
+    updateOpenFortune(null)
+    setIsNewFortuneOpen(false)
+
+    if (checkboxRef.current?.checked) {
+      setIsFortunesFeatureDisabled(true)
+    }
+  }, [updateOpenFortune, checkboxRef, setIsFortunesFeatureDisabled])
+
+  const openFortuneModal = useCallback(() => {
+    if (isDailyFortuneChecked && lastCheckedFortune) {
+      updateOpenFortune(lastCheckedFortune.item)
+    } else {
+      openRandomFortune()
+      setIsNewFortuneOpen(true)
     }
-  }, [isFortuneButtonVisible, lastCheckedFortune, showFortuneButton])
+  }, [isDailyFortuneChecked, openRandomFortune, lastCheckedFortune, updateOpenFortune])
+
+  if (isFortunesFeatureDisabled) return null
 
   return (
     <>
       {openFortune && (
         <FortuneBanner>
-          <FortuneText>{openFortune.text}</FortuneText>
-          <FortuneBannerActions>
-            <ExternalLink href={`https://twitter.com/intent/tweet?text=${twitterText}`}>
-              <SuccessBanner type={'Twitter'}>
-                <span>
-                  <Trans>Share on Twitter</Trans>
-                </span>
-                <SVG src={twitterImage} description="Twitter" />
-              </SuccessBanner>
-            </ExternalLink>
-            <ButtonPrimary onClick={() => updateOpenFortune({ openFortune: null })}>Close</ButtonPrimary>
-          </FortuneBannerActions>
+          <StyledCloseIcon onClick={closeModal}>Close</StyledCloseIcon>
+          <FortuneTitle>
+            {isNewFortuneOpen ? (
+              <>
+                CoW Fortune <i>of the day</i>
+              </>
+            ) : (
+              <>Come back tomorrow for your daily fortune</>
+            )}
+          </FortuneTitle>
+          <FortuneContent>
+            <FortuneText isNewFortuneOpen={isNewFortuneOpen}>{openFortune.text}</FortuneText>
+            <FortuneBannerActions>
+              <StyledExternalLink href={`https://twitter.com/intent/tweet?text=${twitterText}`}>
+                <SuccessBanner type={'Twitter'}>
+                  <span>
+                    <Trans>Share on Twitter</Trans>
+                  </span>
+                  <SVG src={twitterImage} description="Twitter" />
+                </SuccessBanner>
+              </StyledExternalLink>
+              <DontShowAgainBox>
+                <label>
+                  {/*// TODO: tooltip with explanation*/}
+                  <input type="checkbox" ref={checkboxRef} />
+                  <span>Don't show this again</span>
+                </label>
+              </DontShowAgainBox>
+            </FortuneBannerActions>
+          </FortuneContent>
         </FortuneBanner>
       )}
-      {isFortuneButtonVisible && <FortuneButton onClick={openRandomFortune}></FortuneButton>}
+      <FortuneButton onClick={openFortuneModal}>{isDailyFortuneChecked ? '' : '*'}</FortuneButton>
```

Comment:
@fairlighteth can you add please styles for the case, when the daily fortune is not checked? * - is displayed

Reply:
UPD:
https://github.com/cowprotocol/cowswap/pull/2361/commits/b3f6301b5c1cef4e3ed4cded7f07300b5735bf19
I disabled blinking animation when a daily fortune is checked